### PR TITLE
Rat King Names

### LINKED ISSUE #10063: Rat King should have funny names
## Description

I expect the best cheese puns.

**Reproduction**

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR 
rounding out the last of the rat king cruft i've spread over a bunch of prs. This one is just some funny names. We should have support for random metadata supporting multiple datasets.

Closes #10063 
**Screenshots**

**Changelog**

:cl:
- add: The Rat King now can spawn with a multitude of different names. Enjoy!

### CODE REVIEW

File: Resources/Prototypes/Datasets/Names/regalrat.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  - The
+
+- type: dataset
+  id: RegalRatNameTitle
+  values:
+  - King
+  - Lord
+  - Prince
+  - Emperor
+  - Supreme
```
Comment (Visne) [inline Resources/Prototypes/Datasets/Names/regalrat.yml:27]: Wouldn't this be better as an adjective?

File: Resources/Prototypes/Datasets/Names/regalrat.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  - Trash
+  - Garbage
+  - Rat
+  - Vermin
+  - Cheese
+  - Munching
+  - Big
+  - Filth
+  - Sewer
+  - Disposal
```
Comment (Visne) [inline Resources/Prototypes/Datasets/Names/regalrat.yml:16]: Maybe "Disposals"? Up to you.

**Review by Visne — APPROVED** [review]

File: Resources/Prototypes/Datasets/Names/regalrat.yml (hunk 1 above)
Comment (EmoGarbage404) [reply Resources/Prototypes/Datasets/Names/regalrat.yml:27]: this is from the tg list. Personally, i think it works pretty well for things like "cheese supreme" and "filth supreme". I don't really see how something like "supreme lord" ties into the whole rat king vibe.

File: Resources/Prototypes/Datasets/Names/regalrat.yml (hunk 2 above)
Comment (EmoGarbage404) [reply Resources/Prototypes/Datasets/Names/regalrat.yml:16]: "disposals lord" sounds worse than "disposal lord" imo.

Comment (Visne) [reply Resources/Prototypes/Datasets/Names/regalrat.yml:16]: Yeah, sure.

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**